### DISCUSSION TITLE
Menu option to change dir into a recent project

### DISCUSSION THREAD

**Discussion by twiclo** [goolord/alpha-nvim#205] [Q&A] [answered]
I know this isn't exactly alpha related but I figured this is a good place to ask. I'd like a menu option to list the 5 most recent "projects" I've been working on, CD into that directory, then relaunch alpha. I think I'll keep a text file of directories in ~/.cache. Every time a file is saved a `git rev-parse --show-toplevel` will be ran and the cache updated.

Before I set out on making this plugin I wanted to know if this is do-able with something else that already exists. Any suggestions/optimizations to my idea is also welcome

**Comment by goolord** [comment, marked as answer]
1. the startify and theta themes will already update the screen on `DirChanged`, you can test this out by running `:cd ~/foo<CR>` and observing the MRU change.
2. if you can get a list of these projects folders somehow (i'm sure there is some plugin that can provide such an API) then all you would need to do is copy the code for the mru function and sections, change oldfiles to that list, and change the command being run on the filepath to `cd`
3. if you do end up getting something like that to work, lmk and i will upstream it as an extension module similar to the `alpha.term` or `alpha.fortune` module

**Reply by twiclo** [reply]
Can you link me to these modules? I also wrote a function to generate ascii art for the header depending on the name of the git directory you're in: https://asciinema.org/a/55ia2ygqn6KD30GhToZ44F8fX

I've been wondering what the best way to present it to you is. Would you want that as a PR or as one of these modules? Are these modules just separate repos?

**Reply by goolord** [reply]
https://github.com/goolord/alpha-nvim/blob/main/lua/alpha/fortune.lua
https://github.com/goolord/alpha-nvim/blob/main/lua/alpha/term.lua

submit a PR

**Reply by goolord** [reply]
I'm not interested in upstreaming just anything; but a projects menu seems generally useful and that dotfiles ascii art thing seems cool!
